include/mvdtool/bits/sonata_misc.hpp

```diff
@@ -182,11 +192,82 @@ inline std::vector<std::string> SonataFile::getMorphologies(const Range & range)
 }
 
 inline std::vector<std::string> SonataFile::getEtypes(const Range & range) const{
-    return pop_->getAttribute<std::string>(did_etypes, select(range, size_));
-}
+    std::vector<std::string> eTypes;
+    try {
+        mTypes = pop_->getAttribute<std::string>(did_etypes, select(range, size_));
```

Comment:
Only comment here is to try to not have so much copy paste :) The functions are very similar, maybe worth trying to have a generic function

Reply:
Yeah, you are right about this. This was just the first version that I created these functions. I will refactor the code with the proper functions to avoid the repeated code blocks :+1: